Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -181,6 +181,19 @@ public class GraphTraverser: GraphTraversing {
             .first { $0.target.product == .appClip }
     }
 
+    public func buildsForMacCatalyst(path: AbsolutePath, name: String) -> Bool {
+        let dependenciesSupportMacCatalyst = allDependenciesSatisfy(from: .target(name: name, path: path)) { dependency in
+            if let target = self.target(from: dependency) {
+                return target.target.supportsCatalyst
+            } else {
+                // TODO: - Obtain this information from pre-compiled binaries
+                // lipo -info should include "macabi" in the list of architectures
+                return false
+            }
+        }
+        return dependenciesSupportMacCatalyst && (target(path: path, name: name)?.target.supportsCatalyst ?? false)
```

Comment:
nit: maybe we can check `target(path: path, name: name)?.target.supportsCatalyst` at the beginning and return earlier without processing the whole three if false?

Reply:
Very good point. There's no need to traverse the graph if the target we are traversing from doesn't support it. I'll push a fix.